### ISSUE TITLE
Auto Import Related Types for Output Bindings (e.g. Sendgrid)

### ISSUE THREAD

**Issue by JustinGrote** [Azure/azure-functions-powershell-worker#260]
While trying to work with the SendGrid output binding, the JSON example works, but I'd prefer to use the SendGrid C# object because it provides a lot more functionality and a lot less boilerplate work.

However, I found the sendgrid types were not natively available in my session. As a workaround, I did this (in Azure Functions Core Tools):
```powershell
Add-Type -Path C:\Users\JGrote\AppData\Local\Temp\Functions\ExtensionBundles\Microsoft.Azure.Functions.ExtensionBundle\1.0.0\bin\sendgrid.dll -verbose
```
And now I can work with a sendgrid object like the C# example.

Would it be possible to just have these available for ease of use depending on the registered output bindings? At a minimum some higher-class Powershell cmdlets could then be made to wrap around the object and make things like constructing a SendGrid email more structured vs. scraping a raw JSON file together.

**Comment by TylerLeonhardt** [comment]
This is the result of PowerShell function apps running out of process in the PowerShell worker while C# functions run inside the Azure Functions Host process itself. This means that:

* PowerShell functions don't have access to the types you want
* Any object must be serialized to JSON before the PowerShell worker process sends the response back to the Azure Functions Host process... And since we're not checking for this SendGrid type, we just run the object through the same code used for `ConvertTo-Json`.

I think this is one of the scenarios where live object support would be nice and would fix this issue.

We can close this issue to that one... But before we do, we could consider fixing just this issue - if the `Value` of a SendGrid binding is the SendGrid .NET type, we can call `Serialize()` on it... Instead of the normal `ConvertToJson` call.

**Comment by TylerLeonhardt** [comment]
I'd like to see what @AnatoliB thinks

**Comment by JustinGrote** [comment]
@tylerleonhart this was less about the grpc serialization, and more about auto adding the dll assemblies relevant to the loaded output bindings so they don't have to be hunted down like my awkward workaround. Maybe it's too case-by-case, I don't know if the extension bundles have any kind of "hints" in this regard.

**Comment by TylerLeonhardt** [comment]
Right, this issue is two fold though for the complete user experience

1. The types should be available if you add the extension to your Function App
2. You shouldn't have to call a method like `.Serialize()` in order to get an output binding working.

This, to me, is how we properly support types pulled in by func extensions

**Comment by daxian-dbw** [comment]
The `"adding assemblies relevant to declared output binding"` part can be helped with by implementing the `function.proj` support (#173). Then we probably can update the host to generate the appropriate `funciton.proj` that references to the nuget package corresponding to the output bindings declared in `function.json`.

**Comment by JustinGrote** [comment]
Related for Posterity, if you're using the output binding in table storage, it has to be in JSON format, though it comes in from the table as a hashtable (probably because it was auto-converted from JSON), not intuitive but it works.

Basically, my operating philosophy on working with output bindings has been to follow the Javascript example and replace `context.bindings.myOutputBinding.push` with `Push-OutputBinding -Name myOutputBinding`

**Comment by daxian-dbw** [comment]
> if you're using the output binding in table storage, it has to be in JSON format, though it comes in from the table as a hashtable (probably because it was auto-converted from JSON)

I assume the NodeJS worker does the same (auto-convert the JSON string to an object), no?

**Comment by JustinGrote** [comment]
Probably, I have no idea honestly :)

Also confirmed at least for table output that just passing a hash table
works, you don't have to convert to json first (assume the worker does it
on the backend)
